orangecontrib/prototypes/widgets/owhub.py

```diff
@@ -0,0 +1,23 @@
+from PyQt4.QtGui import QSizePolicy
+from Orange.widgets import widget, gui
+
+
+class OWHub(widget.OWWidget):
+    name = "Hub"
+    description = "Universal hub widget"
+    icon = "icons/Hub.svg"
+
+    inputs = [("Object", object, "get_input", widget.Default)]
+    outputs = [("Object", object, widget.Dynamic)]
+
+    NOTHING = "Nothing on input"
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
```

Comment:
OWWidget's init does not have parameters in Orange3.
https://github.com/biolab/orange3/blob/master/Orange/widgets/widget.py#L142

Setting parent (by canvas) is handled in widget's __new__ method.